src/js/wrap.h

```diff
@@ -160,35 +232,61 @@ namespace ccf::js
       return JSWrappedValue(ctx, JS_GetProperty(ctx, val, prop));
     }
 
-    void set(const char* prop, const JSWrappedValue& value) const
+    int set(const char* prop, const JSWrappedValue& value) const
```

Comment:
It looks like most of the uses of this want `std::move` to get the other overload. When do we use this `const&` instance?

Reply:
I don't know, but that's a good question to answer once I'm done making the FFI safe I think.

Another thing that I've put in a pin in for now is that the crypto code (ie. wrappers for OpenSSL) mostly (but not exclusively) throws std::runtime_error to indicate validation failures, and std::logic_error is used for more internal failures. It's important to distinguish them when we validate an x509 chain for example, with some fuzziness. If one of the values in the chain is not a valid PEM for example, is that False or exception (I think it's exception, but it's arguable).

Reply:
> When do we use this const& instance?

The answer was, nowhere legitimate, that's cleaned up now. Separately, I seem to have introduced a double free in historical queries, chasing that now...